go/vt/vtctl/workflow/server.go

```diff
@@ -3310,19 +3344,30 @@ func (s *Server) switchWrites(ctx context.Context, req *vtctldatapb.WorkflowSwit
 		}
 	}
 
-	// Need to lock both source and target keyspaces.
-	tctx, sourceUnlock, lockErr := sw.lockKeyspace(ctx, ts.SourceKeyspaceName(), "SwitchWrites")
+	// Lock the workflow and its source and target keyspaces.
+	lockName := fmt.Sprintf("%s/%s", ts.TargetKeyspaceName(), ts.WorkflowName())
+	ctx, workflowUnlock, lockErr := s.ts.LockName(ctx, lockName, "SwitchWrites")
+	if lockErr != nil {
+		return handleError(fmt.Sprintf("failed to lock the %s workflow", lockName), lockErr)
+	}
+	defer workflowUnlock(&err)
+	// We need to hold the keyspace locks longer than the command timeout.
+	cmdTimeout := req.GetTimeout()
+	if cmdTimeout == nil {
+		cmdTimeout = &vttimepb.Duration{Seconds: 30}
```

Comment:
If this is intended to be a default, the right place to apply it is during flag parsing, no?

Reply:
When coming from vtctldclient, there would always be a value. That does not, however, apply to e.g. unit tests (where I saw the nil reference panic).

Reply:
I tied the default together across vtctldclient and vtctld[server] here: https://github.com/vitessio/vitess/pull/16260/commits/4ba3e580dfc195205450118fc493abadc901c3eb

Reply:
I tied the default together across vtctldclient and vtctld[server] here: https://github.com/vitessio/vitess/pull/16260/commits/90143d5a524bab353e7837ae09a25db1f4586f27